packages/components/htz-components/src/hooks/useWindowSize.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+/* global window */
+import React from 'react';
+import throttle from '../utils/throttle';
+
+const events = new Set<() => void>();
+const onResize = () => events.forEach(fn => fn());
+
+export default function useWindowSize(options: { throttleMs?: number, } = {}) {
+  const { throttleMs = 100, } = options;
+  const isServer = typeof window === 'undefined';
+
+  const [ size, setSize, ] = React.useState({
+    width: isServer ? 0 : window.innerWidth,
+    height: isServer ? 0 : window.innerHeight,
+  });
+
+  const handle = throttle(() => {
+    setSize({
+      width: window.innerWidth,
+      height: window.innerHeight,
+    });
+  }, throttleMs);
+
+  React.useEffect(() => {
+    if (events.size === 0) {
+      window.addEventListener('resize', onResize, true);
+    }
+
+    events.add(handle);
+
+    return () => {
+      events.delete(handle);
+
+      if (events.size === 0) {
+        window.removeEventListener('resize', onResize, true);
+      }
+    };
+  }, [ handle, ]);
```

Comment:
Same as with `useScrollPosition`, no need to detach and reattach every time
```suggestion
  }, [ ]);
```